lib/matplotlib/backend_tools.py

```diff
@@ -626,28 +625,44 @@ def scroll_zoom(self, event):
         # https://gist.github.com/tacaswell/3144287
         if event.inaxes is None:
             return
-        ax = event.inaxes
-        cur_xlim = ax.get_xlim()
-        cur_ylim = ax.get_ylim()
-        # set the range
-        cur_xrange = (cur_xlim[1] - cur_xlim[0])*.5
-        cur_yrange = (cur_ylim[1] - cur_ylim[0])*.5
-        xdata = event.xdata  # get event x location
-        ydata = event.ydata  # get event y location
+
+        scl = self.base_scale
+
         if event.button == 'up':
             # deal with zoom in
-            scale_factor = 1 / self.base_scale
+            direction = 'in'
         elif event.button == 'down':
             # deal with zoom out
-            scale_factor = self.base_scale
+            direction = 'out'
         else:
             # deal with something that should never happen
-            scale_factor = 1
-        # set new limits
-        ax.set_xlim([xdata - cur_xrange*scale_factor,
-                     xdata + cur_xrange*scale_factor])
-        ax.set_ylim([ydata - cur_yrange*scale_factor,
-                     ydata + cur_yrange*scale_factor])
+            scl = 1
+
+        ax = event.inaxes
+        # get the bounding box
+        transD2C = ax.transData.transform
+        xmin, xmax = ax.get_xlim()
+        ymin, ymax = ax.get_ylim()
```

Comment:
I had a short discussion with @QuLogic about this before. The reason for him adding _get_view was to keep the use of the definition of the view outside the axes. So if one created a new class, like polar axes, one could redefine how the _view methods worked and even what they returned without worrying of dependencies. In polar axes, x,y make less sense. This is why I did not use _get_view.

The best compromise I could find is to assume there exists a two way mapping between canvas (x,y) <--> data(x,y) (the transD2C function). I've tested this for polar coordinates as well and it works, but this might not necessarily be true in the future. (The way I see the zoom currently working I'm guessing it treats one axis as r and the other is ignored)

Probably the best option after that would be to have a zoom option in the new _set_view_from_bbox method, as QuLogic suggested.
However, before doing so, I think the definitions of data, axes and canvas coordinates might need to be discussed revisited. Currently, I don't think the axes coordinates are well defined for systems like polar coordinates, documentation [here](http://matplotlib.org/users/transforms_tutorial.html).

Should we just fix it like this for now and make a bigger change later? I feel like modifying _set_view_from_bbox is a bigger change then just rewriting the currently used code (which uses get_xlim() etc).